src/components/Mission.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { joinMission, leaveMission } from '../redux/missions/missions';
+
+const Mission = (props) => {
+  const {
+    mission,
+  } = props;
+  const { id, name, description } = mission;
+
+  const isReserved = (mission) => {
+    if (mission.reserved) return true;
+    return false;
+  };
+  const dispatch = useDispatch();
+  const joinMissionDispatch = (id) => {
+    dispatch(joinMission(id));
+  };
+  const leaveMissionDispatch = (id) => {
+    dispatch(leaveMission(id));
+  };
+  return (
+    <>
+      <td className="font-bold">
+        {name}
+      </td>
+      <td>
+        {description}
+      </td>
+      <td className="align-middle">
+        <span
+          className={isReserved(mission) ? 'badge badge-primary bg-primary' : 'badge badge-secondary bg-secondary'}
+        >
+          {isReserved(mission) ? 'Active Member'.toUpperCase() : 'Not A member'.toUpperCase()}
+        </span>
+      </td>
+      <td className="align-middle">
+        <button
+          onClick={(e) => {
+            e.preventDefault();
+            if (isReserved(mission)) {
+              return leaveMissionDispatch(id);
+            }
+            return joinMissionDispatch(id);
+          }}
+          type="submit"
+          className={isReserved(mission) ? 'btn btn-danger border' : 'btn btn-light border'}
+        >
+          {isReserved(mission) ? 'Leave mission' : 'Join mission' }
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/162113536-4fee3642-4cd9-4baf-b87f-508f39817e25.png)

Therefore:

- [ ] Please, kindly use the React conditional rendering syntax, instead of using ternary operators.

## Explanation

This React rendering syntax, is commonplace in the workplace, and we need to practice that.

## Hints

Everything after the `&&` means "if the left part is true, do the things in the right part".

So in this case:

```JAVASCRIPT
{mission.reserved && ( 
    // render Cancel button
)}
```

It means that if the rocket is reserved is TRUE, then it will do the thing on the right (...) after the `&&`.

You can build if, then, else schemas by using `!!`:

```JAVASCRIPT
{!!mission.reserved && ( 
    // don't render the cancel button
)}
```

_Note: This is just an example to give you an idea of how to make the solution work.  You should use this as a mini-tutorial and implement what you need, according to your own project needs.  This is just the result of my own personal experience and effort, and it is by no means to be taken as an "official" solution._

I hope this is helpful to you! 💪